### ISSUE TITLE
Random failed builds cause of uglify

### ISSUE THREAD

**Issue by IagoSRL** [loconomics/loconomics#839]
### Long Description
Several failed builds were registered at Travis CI last week, sometimes even for changes not related with the front-end (that is the only part that is build at Travis right now) when previous commit build correctly. 

<strike>Building locally, Windows 10 machine, was fine on all that cases.</strike> Fails at local builds started to appear, see https://github.com/loconomics/loconomics/issues/839#issuecomment-373692147

Examples:
- https://travis-ci.org/loconomics/loconomics/builds/351554897
- https://travis-ci.org/loconomics/loconomics/builds/352966484
- https://travis-ci.org/loconomics/loconomics/builds/351554897

It seems all errors happens when running uglify. We use it through the package `grunt-contrib-uglify-es`, pointing to a specific commit.

I re-enabled Yarn at Travis just in case was related with some different version, but is still happening.

**Comment by IagoSRL** [comment]
First time this error happened to me in local environment, details:
- At commit fe5e9f5e3
- Same error source: uglify-es
- Same error message: `Unexpected token eof «undefined», expected punc «,». `
- Same file: `app.js`.
- Different line number but that's obvious because of the changes in the file (line 29465, col 5, pos 1049296, versus line: 29446, col: 5, pos: 1046037 in the Travis builds included in the initial comment)
- Command `yarn grunt build-web-release`
- **Trying same build command again, 1 second after, changing nothing, works fine**
- There was a recent upgrade on Browserify, Babel, and some dependencies. Initial errors happened with different versions.
- It's still not clear if is a problem with Uglify or with the generation of the JS bundle (Browserify, Babel and some transforms involved). **Next time must try running just the uglify command**.

Updating the issue title since is not a problem specific with builds at Travis CI anymore.

**Comment by IagoSRL** [comment]
I committed a change to the default build (the one is being run by Travis) to prevent duplication of tasks (because of reusing aliases at Grunt that internally trigger some same sub-tasks, specially some heavy ones), and added lost build steps for preparing a Phonegap release build, so gets tested. The task (`grunt` == `yarn run build`) takes now around 43% the time of the original.

As part of that, run multiple times at local the build process, as a whole and separated parts (browserify and uglify ones specially) lot of times and didn't get errors. Both before doing the task change and after.
It seems to happens more frequently at Travis than in local, were actually that commit I pushed failed! just after test the build of that exact commit around 10 times in local, all of them ended successfully.

It's very confusing what is triggering this random fails, but I was thinking about to use again the main release of uglify rather than the 'harmony / ES' branch, since actually we are processing with Babel (and targeting some old browsers, like IE) so the result that Uglify receives should have not ES6+ syntax and then should not suppose a problem. But really, couldn't detect still the fail is there.
